fix that returns empty string when a segwit add is found

alejoacosta74 opened the pull request:
When calling `eth_getBlockByNumber` on a block that contains a tx with a `segwit` address, Janus will crash

```
echo: http: panic serving 186.108.222.16:61740: runtime error: slice bounds out of range [:21] with capacity 0
goroutine 249173 [running]:
net/http.(*conn).serve.func1(0xc0002620a0)
    /usr/local/go/src/net/http/server.go:1800 +0x139
panic(0xb5ee80, 0xc00014bbe0)
    /usr/local/go/src/runtime/panic.go:975 +0x3e3
github.com/qtumproject/janus/pkg/utils.ConvertQtumAddress(0xc0000c2a20, 0x2a, 0x40, 0x810300, 0xc0004ee000, 0x0)
    /go/src/github.com/qtumproject/janus/pkg/utils/hex.go:52 +0x1b8
```

This fix filters out the `segwit` address and returns an empty string `""`

ripply commented:
testnet/mainnet will have different prefixes I believe

https://gist.github.com/matthewdowney/9e556fc1a3ad5d331f388062163decc5#file-gen-py-L25

can you just check if the base58.Decode returns an size 0 result and error that way

alejoacosta74 commented:
I have added a checkDecode() that returns an err if the address cannot be decoded.  Also I added a unit test for `ConvertQtumAddress`:

```
	_, _, err := base58.CheckDecode(address)
	if err != nil {
		return "", errors.Errorf("invalid address")
	}
```
